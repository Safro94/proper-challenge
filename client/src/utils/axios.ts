import axios, { AxiosError, AxiosRequestConfig } from 'axios';
import https from 'https';

const opts: AxiosRequestConfig = {
	httpsAgent: new https.Agent({ rejectUnauthorized: false }),
};

const interceptorError = (error: AxiosError) => Promise.reject(error.response);

const instance = axios.create(opts);

instance.interceptors.response.use(response => response, interceptorError);
instance.defaults.headers.common['x-api-key'] = process.env.REACT_APP_API_KEY;

export default instance;
