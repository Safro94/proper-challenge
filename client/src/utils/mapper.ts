import { IAddress, IDawaResponse } from '../types';

const mapDawaResponse = (data: IDawaResponse[]): IAddress[] => {
	return data.map(item => ({
		text: item.tekst,
		type: item.type,
	}));
};

export { mapDawaResponse };
