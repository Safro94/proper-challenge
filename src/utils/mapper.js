const mapProperties = properties =>
	properties.map(({ address, size, rooms, utilities, tenantName }) => ({
		address,
		size,
		rooms,
		utilities,
		tenantName,
	}));

module.exports = { mapProperties };
