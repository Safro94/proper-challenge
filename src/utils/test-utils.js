const mockReq = params => {
	const req = { ...params };
	return req;
};

const mockRes = params => {
	const res = { ...params };
	res.json = jest.fn().mockReturnValue(res);
	res.status = jest.fn().mockReturnValue(res);

	return res;
};

module.exports = {
	mockReq,
	mockRes,
};
