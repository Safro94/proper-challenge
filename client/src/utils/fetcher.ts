import { RequestMethods, RequestStatus } from '../types';

import axios from './axios';

interface IFetcher {
	method?: RequestMethods;
	url: string;
	data?: null | any;
	body?: null | any;
	headers?: null | any;
}

interface IResponse {
	data: any;
	status: RequestStatus;
}

const fetcher = async ({
	method = RequestMethods.GET,
	url,
	body = null,
	headers = null,
}: IFetcher): Promise<IResponse> => {
	let data;
	let status = RequestStatus.Pending;

	try {
		const result = await axios[method](
			url,
			JSON.parse(headers),
			JSON.parse(body)
		);
		data = result.data;
		status = RequestStatus.Resolved;
	} catch (error) {
		status = RequestStatus.Rejected;
		throw error;
	}

	return { data, status };
};

export default fetcher;
