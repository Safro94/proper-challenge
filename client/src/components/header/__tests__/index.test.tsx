import { render, screen } from '../../../utils/test-utils';

import Header from '..';
import userEvent from '@testing-library/user-event';

describe('Header', () => {
	const handleSidebar = jest.fn();
	const isOpen = false;

	beforeEach(() => {
		render(<Header handleSidebar={handleSidebar} isOpen={isOpen} />);
	});

	it('should match the snapshot', () => {
		const { container } = render(
			<Header handleSidebar={handleSidebar} isOpen={isOpen} />
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render correctly', () => {
		expect(screen.getByTestId('icon')).toBeInTheDocument();
		expect(screen.getByRole('link')).toBeInTheDocument();
		expect((screen.getByRole('link') as HTMLLinkElement).href).toBe(
			'http://localhost/'
		);
		expect(screen.getByText(/welcome matias/i)).toBeInTheDocument();
		expect(screen.getByAltText(/avatar/i)).toBeInTheDocument();
	});

	it('should call handleSidebar when the menu is clicked', () => {
		expect(handleSidebar).not.toHaveBeenCalled();
		userEvent.click(screen.getByTestId('menu'));

		expect(handleSidebar).toHaveBeenCalledTimes(1);
		expect(handleSidebar).toHaveBeenCalledWith(!isOpen);
	});
});
