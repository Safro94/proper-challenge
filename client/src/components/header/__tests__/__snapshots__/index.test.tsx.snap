// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  class="sc-dlnjwi hAvndu"
>
  <div
    class="sc-hKFxyN effxTH"
  >
    <div
      class="sc-eCApnc iXMvPy"
    >
      <div
        class="sc-jSFjdj kcHInj"
        data-testid="menu"
      >
        <svg
          data-testid="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </div>
      <div
        class="sc-gKAaRy fzAOkR"
      >
        <a
          class="sc-iCoGMd iaZHzH"
          href="/"
        >
          <svg>
            logo.svg
          </svg>
        </a>
      </div>
    </div>
    <div
      class="sc-fujyAs islHaU"
    >
      <span
        class="sc-jrsJWt fPUaef"
      >
        welcome
         
        Matias
      </span>
      <img
        alt="Avatar"
        class="sc-pNWdM hQKEDH"
        src="avatar.jpeg"
      />
      <div
        class="sc-bdnxRM iJoJsd"
      >
        <button
          class="sc-gtsrHT gLvTEs"
        >
          EN
        </button>
        <button
          class="sc-gtsrHT bIpGJm"
        >
          ES
        </button>
      </div>
    </div>
  </div>
</div>
`;
